src/common/mesh_utils.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "mesh_utils.hpp"
+
+#include "common/logger.hpp"
+#include "fmt/fmt.hpp"
+
+namespace serac {
+
+std::shared_ptr<mfem::ParMesh> buildParallelMesh(const std::string& mesh_file, const int refine_serial,
```

Comment:
This belongs in a separate PR, but I think we can use a `unique_ptr<mfem::ParMesh>` if not just an `mfem::ParMesh` here and in the `FiniteElementState` struct.  Either way this function will make that refactoring a lot easier. :+1:

Reply:
I've thought a bit about this, and the problem is that each `FiniteElementState` needs access to the mesh. Certain physics algorithms actually change the mesh (via AMR or element erosion), so a `const Parmesh&` is not really valid. It becomes a question of who would "own" the mesh. I've considered putting it in the "vault-type" problem level object which owns a unique pointer, but that is definitely a separate PR.